parsec/mca/device/device_gpu.c

```diff
@@ -2342,7 +2333,17 @@ parsec_device_kernel_epilog( parsec_device_gpu_module_t *gpu_device,
 
         assert( 0 <= gpu_copy->readers );
 
-        if( gpu_task->pushout & (1 << i) ) {
+        if (cpu_copy->flags & PARSEC_DATA_FLAG_DISCARDED) {
+            PARSEC_DEBUG_VERBOSE(20, parsec_gpu_output_stream,
```

Comment:
Here is where the race-condition could happen.

Reply:
I think this is not a big issue here since we're not modifying the flags and neither the data_t nor the host copy will be freed, since the device copy maintains its reference. The worst thing that can happen is that we miss the flag being set and the release is delayed until we try to evict.